src/services/swap_service.ts

```diff
@@ -333,6 +334,15 @@ export class SwapService {
             shouldGenerateQuoteReport,
             shouldIncludePriceComparisonsReport: !!includePriceComparisons,
             samplerMetrics: SAMPLER_METRICS,
+            fillAdjustor:
+                enableSlippageProtect && this.slippageModelManager
```

Comment:
if the API param is passed and enabled, or the default is to enable it globally, we use a `SlippageModelFillAdjustor`. 

If not we just default to the `IdentityFillAdjustor` which makes no modifications.